src/main/java/persistence/sql/context/impl/DefaultEntityPersister.java

```diff
@@ -22,31 +22,39 @@
 import java.util.List;
 import java.util.logging.Logger;
 
-public class DefaultEntityPersister implements EntityPersister {
+public class DefaultEntityPersister<T> implements EntityPersister<T> {
     private static final Logger logger = Logger.getLogger(DefaultEntityPersister.class.getName());
     private final Database database;
     private final NameConverter nameConverter;
-    private final MetadataLoader<?> metadataLoader;
+    private final MetadataLoader<T> metadataLoader;
 
-    public DefaultEntityPersister(Database database, NameConverter nameConverter, MetadataLoader<?> metadataLoader) {
+    public DefaultEntityPersister(Database database, NameConverter nameConverter, MetadataLoader<T> metadataLoader) {
         this.database = database;
         this.nameConverter = nameConverter;
         this.metadataLoader = metadataLoader;
     }
 
     @Override
-    public Object insert(Object entity) {
+    public T insert(Object entity) {
```

Comment:
```suggestion
    public T insert(T entity) {
```
제네릭을 활용하도록 바꾸어주셨네요 :) 파라미터도 Object에서 변경해보면 조금 더 타입안정성을 강화할 수 있을 것 같아요.